Add technologies used in Talawa

Yogita-T opened the pull request:
Added a section called **Technologies Used** to the About Us page in Talawa-docs.

Please review the changes.

Yogita-T committed:
Add technologies used in Talawa

Signed-off-by: Yogita-T <2020a1r118@mietjammu.in>

palisadoes commented on docs/about.md:
```diff
@@ -34,3 +34,22 @@ Talawa’s desired main features are:
+
+4. Security: Talawa uses security technology like SSL (Secure Socket Layer) which helps to create a secure encrypted connection between the server(our project) and the client and protects Talawa from hacking and other security threats.
+
+5. Payment Processing: Talawa use payment processing technology like Razorpay to handle payments.
```
We plan to have plugins that will interact with payment processing gateways.

palisadoes commented on docs/about.md:
```diff
@@ -34,3 +34,22 @@ Talawa’s desired main features are:
+
+3. Database Management System: Talawa uses database management system like Python SQLAlchemy as an ORM(Object–relational mapping) and MongoDB is used to store and manage the website's data.
+
+4. Security: Talawa uses security technology like SSL (Secure Socket Layer) which helps to create a secure encrypted connection between the server(our project) and the client and protects Talawa from hacking and other security threats.
```
Our  project ideas include various encryption technologies including, but not limited to SSL

palisadoes commented on docs/about.md:
```diff
@@ -34,3 +34,22 @@ Talawa’s desired main features are:
+
+2. Back-end Technologies: Talawa uses server-side scripting languages like GraphQL, React and Node.js to handle the website's back-end functionality.
+
+3. Database Management System: Talawa uses database management system like Python SQLAlchemy as an ORM(Object–relational mapping) and MongoDB is used to store and manage the website's data.
```
Talawa API uses Mongo DB with the mongoose ORM . Please reword this.

palisadoes commented on docs/about.md:
```diff
@@ -34,3 +34,22 @@ Talawa’s desired main features are:
+
+1. Front-end Technologies: HTML, CSS, Flutter, TypeScript and JavaScript is used to create the website's front-end user interface.
+
+2. Back-end Technologies: Talawa uses server-side scripting languages like GraphQL, React and Node.js to handle the website's back-end functionality.
```
We don't use Node.js any more. We replaced it with Typescript. React isn't a backend technology.

palisadoes commented on docs/about.md:
```diff
@@ -34,3 +34,22 @@ Talawa’s desired main features are:
+Technology plays an increasingly significant role in improving access to education for people living in impoverished areas and developing countries. Some of the technologies that are being used by Talawa are:
+
+
+1. Front-end Technologies: HTML, CSS, Flutter, TypeScript and JavaScript is used to create the website's front-end user interface.
```
We have a web portal and mobile app. The technologies for each specifically need to be mentioned.

Yogita-T replied:
Could you please help me to understand this thing a little briefly?

palisadoes replied:
For example split it somehow between web front end technologies and mobile app front end technologies.

Yogita-T replied:
Done

palisadoes commented on docs/about.md:
```diff
@@ -34,3 +34,22 @@ Talawa’s desired main features are:
+
+6. Cloud Hosting: The foundation may use a cloud hosting service like Amazon Web Services (AWS), Microsoft Azure, or Google Cloud Platform (GCP) to host Talawa.
+
+7. Content Management System (CMS): Talawa is built on a CMS like WordPress, Joomla, or Drupal to manage and update its content.
```
Talawa is built on the technologies above. We use Docusaurus for our supporting documentation.

Yogita-T committed:
Upadted about.md

Yogita-T commented:
@palisadoes Could you review the changes, please? Thankyou.

Yogita-T commented:
@palisadoes awaiting review!

kayceeDev commented on docs/about.md:
```diff
@@ -34,3 +34,21 @@ Talawa’s desired main features are:
+
+Technology plays an increasingly significant role in improving access to education for people living in impoverished areas and developing countries. Some of the technologies that are being used by Talawa are:
+
+1. Front-end Technologies: HTML, CSS, and JavaScript (React) is used to create the website's front-end user interface. And, Flutter is used to create the mobile application. 
```
Can you add the repo which uses these technologies?

kayceeDev commented on docs/about.md:
```diff
@@ -34,3 +34,21 @@ Talawa’s desired main features are:
+
+1. Front-end Technologies: HTML, CSS, and JavaScript (React) is used to create the website's front-end user interface. And, Flutter is used to create the mobile application. 
+
+2. Back-end Technologies: Talawa uses server-side scripting languages like GraphQL and TypeScript to handle the website's back-end functionality.
```
Same here too

kayceeDev commented on docs/about.md:
```diff
@@ -34,3 +34,21 @@ Talawa’s desired main features are:
+
+### Technologies Used
+
+Technology plays an increasingly significant role in improving access to education for people living in impoverished areas and developing countries. Some of the technologies that are being used by Talawa are:
```
The first sentence is exaggerated. We should be more concise and straightforward since Talawa features have been discussed already in the docs.

kayceeDev reviewed (changes requested):
Overall, we should be straightforward with the technologies used as well as make sure we outline the concerned repositories. Well done so far.

salman-farooqi commented on docs/about.md:
```diff
@@ -34,3 +34,21 @@ Talawa’s desired main features are:
+
+3. Database Management System: Talawa API uses database management system like MongoDB with the mongoose ORM (Object–relational mapping) which is used to store and manage the website's data.
+
+4. Security: Talawa uses various encryption technologies to protect sensitive data like SSL (Secure Socket Layer) which helps to create a secure encrypted connection between the server(our project) and the client and protects Talawa from hacking and other security threats.
```
Use ( 'space' and 'comma' ) between " `server (our project) and the client, and protects Talawa...` "

Yogita-T replied:
Noted! @salman-farooqi

Yogita-T commented:
![technologiesUsed](https://user-images.githubusercontent.com/76475512/225024565-9ef01884-a3df-4bf2-9e25-c4dc99e2fa8a.jpg)
* Added links to the respective repos along with the technologies used in them. 
* Changed the first line and made it more concise, as suggested by @kayceeDev .
* Used  ( 'space' and 'comma' ) between  @salman-farooqi.
Please review the changes again! Sorry for any inconvenience caused.

Yogita-T committed:
Added the repositories links to various projects.

Made the suggested changes as well.

Yogita-T committed:
Added the links of repositories.

Made the suggested changes.

Yogita-T commented:
@Cioppolo14  Could you review the changes?

Yogita-T commented:
Any reviews?

Cioppolo14 commented:
@Yogita-T Please resolve palisadoes' comments.

Yogita-T commented:
> @Yogita-T Please resolve palisadoes' comments.

I have already resolved the comments.

Cioppolo14 commented:
@Yogita-T It is still showing that Palisadoes comments aren't addressed. I looked at them, and it does appear they are, but can you see on your end if there is something you need to do to document that?

Yogita-T commented:
> @Yogita-T It is still showing that Palisadoes comments aren't addressed. I looked at them, and it does appear they are, but can you see on your end if there is something you need to do to document that?

I looked at them as well, looks like I have addressed all of them. I have re-requested a review from @palisadoes.

palisadoes commented on docs/about.md:
```diff
@@ -34,3 +34,22 @@ Talawa’s desired main features are:
+
+1. Front-end Technologies: HTML, CSS, and JavaScript (React) is used to create the website's front-end user interface. And, Flutter is used to create the mobile application.
+2. Back-end Technologies: Talawa uses server-side scripting languages like GraphQL and TypeScript to handle the website's back-end functionality.
+3. Database Management System: Talawa API uses database management system like MongoDB with the mongoose ORM (Object–relational mapping) which is used to store and manage the website's data.
```
Remove `like`. We use it.

palisadoes commented on docs/about.md:
```diff
@@ -34,3 +34,22 @@ Talawa’s desired main features are:
+1. Front-end Technologies: HTML, CSS, and JavaScript (React) is used to create the website's front-end user interface. And, Flutter is used to create the mobile application.
+2. Back-end Technologies: Talawa uses server-side scripting languages like GraphQL and TypeScript to handle the website's back-end functionality.
+3. Database Management System: Talawa API uses database management system like MongoDB with the mongoose ORM (Object–relational mapping) which is used to store and manage the website's data.
+4. Security: Talawa uses various encryption technologies to protect sensitive data like SSL (Secure Socket Layer) which helps to create a secure encrypted connection between the server (our project) and the client, and protects Talawa from hacking and other security threats.
```
Remove `like` replace with `including`. Mention that adding SSL was a GSoC 2023 idea.

palisadoes commented on docs/about.md:
```diff
@@ -34,3 +34,22 @@ Talawa’s desired main features are:
+2. Back-end Technologies: Talawa uses server-side scripting languages like GraphQL and TypeScript to handle the website's back-end functionality.
+3. Database Management System: Talawa API uses database management system like MongoDB with the mongoose ORM (Object–relational mapping) which is used to store and manage the website's data.
+4. Security: Talawa uses various encryption technologies to protect sensitive data like SSL (Secure Socket Layer) which helps to create a secure encrypted connection between the server (our project) and the client, and protects Talawa from hacking and other security threats.
+5. Payment Processing: Talawa uses payment processing technology like Razorpay to handle payments. Talawa is also planning to have plugins that will interact with payment processing gateways. Plugins will interact with payment processing gateways by integrating the gateway's API into the plugin. This will allow the plugin to communicate with the payment gateway and process payments securely.
```
We don't use Razor pay. Mention that a GSoC 2023 project is to allow access to multiple payment processor gateways. Access to each gateway would be activated via a dedicated plugin.

palisadoes commented on docs/about.md:
```diff
@@ -34,3 +34,22 @@ Talawa’s desired main features are:
+3. Database Management System: Talawa API uses database management system like MongoDB with the mongoose ORM (Object–relational mapping) which is used to store and manage the website's data.
+4. Security: Talawa uses various encryption technologies to protect sensitive data like SSL (Secure Socket Layer) which helps to create a secure encrypted connection between the server (our project) and the client, and protects Talawa from hacking and other security threats.
+5. Payment Processing: Talawa uses payment processing technology like Razorpay to handle payments. Talawa is also planning to have plugins that will interact with payment processing gateways. Plugins will interact with payment processing gateways by integrating the gateway's API into the plugin. This will allow the plugin to communicate with the payment gateway and process payments securely.
+6. Cloud Hosting: The foundation may use a cloud hosting service like Amazon Web Services (AWS), Microsoft Azure, or Google Cloud Platform (GCP) to host Talawa.
```
Mention that this is a GSoC 2023 idea to include support for cloud hosting services including the ones mentioned.

palisadoes reviewed (changes requested):
Please make the suggested changes.

Yogita-T commented:
Closing this as I created a new PR. Thank you!
